File: plugins/sift/ios/Classes/SiftPlugin.m
#import "SiftPlugin.h"
#import "Sift/Sift.h"

@implementation SiftPlugin
+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
  FlutterMethodChannel* channel = [FlutterMethodChannel
      methodChannelWithName:@"sift"
            binaryMessenger:[registrar messenger]];
  SiftPlugin* instance = [[SiftPlugin alloc] init];
  [registrar addMethodCallDelegate:instance channel:channel];
}

- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
  if ([@"getPlatformVersion" isEqualToString:call.method]) {
    result([@"iOS " stringByAppendingString:[[UIDevice currentDevice] systemVersion]]);
  } else if ([@"setSiftConfig" isEqualToString:call.method]) {
    NSString *accountId = call.arguments[@"accountId"];
    NSString *beaconKey = call.arguments[@"beaconKey"];

    Sift *sift = [Sift sharedInstance];
    [sift setAccountId:accountId];
    [sift setBeaconKey:beaconKey];
    [sift upload];

    result([@"Success sift"]);
  } else if ([@"setUserID" isEqualToString:call.method]) {
    NSString *userId = call.arguments[@"id"];

    [[Sift sharedInstance] setUserId:userId];
  } else if ([@"unsetUserID" isEqualToString:call.method]) {
    [[Sift sharedInstance] unsetUserId];
  } else {
    result(FlutterMethodNotImplemented);
  }
}

@end

File: plugins/sift/ios/Classes/SiftPlugin.swift
/*
import Flutter
import UIKit
import Sift

public class SiftPluginImpl: NSObject, FlutterPlugin {

  var channel: FlutterMethodChannel

  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "sift", binaryMessenger: registrar.messenger())
    let instance = SiftPlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  init(channel: FlutterMethodChannel) {
    self.channel = channel
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    let channel = self.channel

    switch call.method {
      case "getPlatformVersion":
        result("iOS " + UIDevice.current.systemVersion)
      default:
        result(FlutterMethodNotImplemented)
    }
  }
}
*/